### PR TITLE
Set correct async/generator in IIFE for params

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Current behavior:
```js
function* f(a = 1) {
  var a = yield;
}
```

```js
function* f() {
  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
  return function (a) {
    var a = yield;
  }(a);
}
```

### CODE REVIEW

File: packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+function f() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
+  return function* (a) {
+    var a = yield a;
+    return a;
+  }(a);
+}
+
+function g() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
```
Comment (jridgewell) [inline packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js]: The default expression could actually throw a sync error, which needs to be caught sand turned into a rejected promise.

File: packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function f() {
+  let a = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 1;
```
Comment (jridgewell) [inline packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js]: Similar problem here, I don’t think a generator would throw until the first `next` call. I may be wrong though.

Comment (nicolo-ribaudo) [reply packages/babel-plugin-transform-parameters/test/fixtures/regression/scope-gen-async/output.js]: Generators throw immediatly (`(function* (x = a.b) {})()` throws)

**Comment by nicolo-ribaudo** [comment]
@ulrichb This fixes your error about `await` outside of an async function.

**Comment by ulrichb** [comment]
@nicolo-ribaudo Thank you very much!

**Comment by nicolo-ribaudo** [comment]
The problem is that we don't have a place where we can put a `var _this = this` declaration.

**Pull request merged**